modules/library/main/src/test/java/org/geotools/data/collection/ListFeatureCollectionTest.java

```diff
@@ -131,6 +141,65 @@ public void iterator() {
         assertTrue(copy.isEmpty());
     }
 
+    @Test
+    public void removeAndAddFeatureBounds() {
```

Comment:
please cite bug number in javadoc; it helps with traceability.

This is a suggestion only, something we try and do to help others.